cli/azd/pkg/tools/docker/docker.go

```diff
@@ -90,11 +93,26 @@ func (d *docker) Build(
 
 	args = append(args, buildContext)
 
-	res, err := d.executeCommand(ctx, cwd, args...)
+	// Build and produce output
+	previewer := d.console.ShowPreviewer(ctx,
+		&input.ShowPreviewerOptions{
+			Prefix:       "  ",
+			MaxLineCount: 8,
+			Title:        "Docker Output",
+		})
+	runArgs := exec.NewRunArgs("docker", args...).WithCwd(cwd).WithStdOut(previewer)
```

Comment:
I'm not a big fan that now this docker CLI wrapper now needs to have dependencies directly to the console and manages a UX preview experience.  

In my opinion the tool should remain a tool and the UX concerns should be handled at a higher level component closes to our commands/actions.

Reply:
Ultimately something that I could see happening is that our `CommandRunner` broadcasts messages/events when it is invoking a command.  Then our UX layer can capture these commands, filter out the commands that we wish to take action on and attach the preview experience keeping our tool layer clean and loosely coupled.